libstuff/SQResult.cpp

```diff
@@ -183,3 +188,55 @@ SQResult& SQResult::operator=(const SQResult& other) {
     }
     return *this;
 }
+
+vector<SQResultRow>::iterator SQResult::begin() {
+    return rows.begin();
+}
+
+vector<SQResultRow>::iterator SQResult::end() {
+    return rows.end();
+}
+
+vector<SQResultRow>::const_iterator SQResult::begin() const {
+    return rows.begin();
+}
+
+vector<SQResultRow>::const_iterator SQResult::end() const {
+    return rows.end();
+}
+
+vector<SQResultRow>::const_iterator SQResult::cbegin() const {
+    return rows.cbegin();
+}
+
+vector<SQResultRow>::const_iterator SQResult::cend() const {
+    return rows.cend();
+}
+
+void SQResult::resize(size_t newSize) {
+    rows.resize(newSize);
+    for (auto& row : rows) {
+        row.result = this;
+    }
+}
+
+void SQResult::push_back(const SQResultRow& row) {
+    SQResultRow newRow = row;
+    newRow.result = this;
+    rows.push_back(newRow);
+}
+
+void SQResult::emplace_back(const SQResultRow& row) {
+    SQResultRow newRow = row;
+    newRow.result = this;
+    rows.emplace_back(newRow);
+}
+
+SQResultRow& SQResult::back() {
+    return rows.back();
+}
```

Comment:
This is another non-const accessor that I'd prefer not to have if we can get away with it, which maybe we can if it's only used by MySQL.cpp or internally in bedrock.